test/llvmcall.jl

```diff
@@ -48,3 +48,44 @@ baremodule PlusTest
     end
     @test Int32(1)+Int32(2)==Int32(3)
 end
+
+# Test whether declarations work properly
+function undeclared_ceil(x::Float64)
+    llvmcall("""%2 = call double @llvm.ceil.f64(double %0)
+        ret double %2""", Float64, Tuple{Float64}, x)
+end
+@test_throws ErrorException undeclared_ceil(4.2)
```

Comment:
The test seems to be quite finicky

Executing from the REPL:
````
julia> function undeclared_ceil(x::Float64)
           Base.llvmcall("""%2 = call double @llvm.ceil.f64(double %0)
               ret double %2""", Float64, Tuple{Float64}, x)
       end
undeclared_ceil (generic function with 1 method)

julia> undeclared_ceil(4.2)
ERROR: error compiling undeclared_ceil: Failed to parse LLVM Assembly: 
julia: <string>:3:18: error: use of undefined value '@llvm.ceil.f64'
%2 = call double @llvm.ceil.f64(double %0)
````

````
try 
         undeclared_ceil(4.3)
       catch err
         println(err)
       end
ErrorException("error compiling undeclared_ceil: Failed to parse LLVM Assembly: \njulia: <string>:3:18: error: use of undefined value '@llvm.ceil.f64'\n%2 = call double @llvm.ceil.f64(double %0)\n                 ^\n")
````

````
julia> Test.@test_throws ErrorException undeclared_ceil(4.3)

julia> Test.@test_throws ErrorException undeclared_ceil(4.3)
ERROR: test failed: undeclared_ceil(4.3) did not throw ErrorException
 in expression: undeclared_ceil(4.3)
 in error at error.jl:21
 in default_handler at test.jl:29
 in do_test_throws at test.jl:74
````

````
julia> try 
         undeclared_ceil(4.3)
       catch err
         println(typeof(err))
       end
ErrorException
````